### PR TITLE
feat(oas32): add basic OpenAPI 3.2.0 support

### PR DESCRIPTION
**Pull request by robert-hebel-sb**
## Add basic OpenAPI 3.2.0 support                                                                                                                                                                                                                                                   
   
  ### Description                                                                                                                                                                                                                                                                      
                                                            
  This PR adds basic support for OpenAPI Specification 3.2.0 to Swagger UI. The implementation focuses on the foundational features introduced in OAS 3.2:

  **Core Features:**
  - ✅ **QUERY HTTP method support** - Implements the new QUERY operation per draft-ietf-httpbis-safe-method-w-body
  - ✅ **Info.summary field** - Renders the short summary field in the Info Object
  - ✅ **Version detection** - Properly detects and handles OAS 3.2.x specifications

  **Implementation Details:**
  - New `oas32` plugin following the established plugin architecture
  - QUERY operations integrated into the operations selector pipeline
  - Distinctive purple/magenta color (#9D408A) for QUERY operations to differentiate from GET
  - Dark mode support with carefully selected color scheme for QUERY operations
  - OAS 3.2 specs are properly recognized as OAS 3.x (inheriting servers, security, etc.)
  - Selector wrappers ensure QUERY operations are included in operation lists

  **Enhanced Features (moved to future PR):**
  The following OAS 3.2 features are excluded from this basic implementation and will be addressed in a separate enhancement PR:
  - `$self` field for base URI resolution
  - `additionalOperations` for custom HTTP methods
  - `mediaTypes` in Components Object
  - `pathItems` in Components Object
  - Tag enhancements (summary, kind, parent)
  - `querystring` parameter location
  - `itemSchema` for streaming responses

  ### Motivation and Context

  OpenAPI 3.2.0 was released as the latest version of the OpenAPI Specification. Swagger UI needs to support this version to maintain compatibility with the evolving API ecosystem.

  This PR provides the essential foundation for OAS 3.2 support, focusing on the most commonly used features (QUERY operations and info.summary) that users will encounter first. The phased approach allows for:
  1. Early adoption of basic OAS 3.2 features
  2. Thorough testing of the core functionality
  3. Incremental addition of advanced features in future PRs

  The QUERY HTTP method is particularly important as it addresses a common need for search/query operations with request bodies, following the IETF draft standard.

  ### How Has This Been Tested?

  **Unit Tests:**
  - ✅ All 41 unit tests passing across 6 test suites
  - Tests for OAS32 plugin selectors, wrappers, and components
  - Contact and license field tests
  - QUERY operation selector tests

  **E2E Tests (Cypress):**
  - ✅ `oas32-component-only.cy.js` - Verifies component-only specs render without errors
  - ✅ `oas32-version-detection.cy.js` - Confirms OAS 3.2.0 detection and info.summary rendering
  - ✅ `oas32-query-operation.cy.js` - Comprehensive QUERY operation rendering verification
  - ✅ `oas32-contact-and-license.cy.js` - Contact and license field rendering tests
  - ✅ `oas-badge.cy.js` - Updated to include OAS 3.2.0 badge test

  **Manual Testing:**
  - Tested with multiple OAS 3.2.0 specifications
  - Verified QUERY operations render correctly in both light and dark modes
  - Confirmed info.summary field displays properly
  - Tested that OAS 3.2 specs inherit OAS 3.x features (servers, security schemes, etc.)
  - Verified color distinction between QUERY and GET operations

  **Testing Environment:**
  - Node.js 22.11.0 (as specified in `.nvmrc`)
  - macOS (Darwin 24.6.0)
  - Tested in Chrome, Firefox, and Safari (via Cypress)

  ### Screenshots (if appropriate):

  **QUERY Operation (Light Mode):**
  The QUERY operation displays with a distinctive purple/magenta color (#9D408A) that clearly differentiates it from GET operations.

<img width="1664" height="837" alt="Screenshot 2026-02-10 at 16 00 37" src="https://github.com/user-attachments/assets/2453527d-a271-4360-9fef-2e3d9c6b9ea7" />
<img width="1533" height="1077" alt="Screenshot 2026-02-10 at 16 00 47" src="https://github.com/user-attachments/assets/2b88342e-ffd6-424f-b92a-0b7a65a215d5" />


  **QUERY Operation (Dark Mode):**
  Dark mode uses a coordinated purple color scheme with good contrast for readability.

<img width="1629" height="1025" alt="Screenshot 2026-02-10 at 16 00 13" src="https://github.com/user-attachments/assets/7ee5de2e-6820-44e1-adbc-a5be24aa9309" />
<img width="1677" height="802" alt="Screenshot 2026-02-10 at 16 00 23" src="https://github.com/user-attachments/assets/37560d05-3821-4e91-abee-354feeaa6629" />


  **Info Summary Field:**
  The info.summary field renders as a short summary line below the API title.

  ## Checklist

  ### My PR contains...
  - [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
  - [ ] Dependency changes (any modification to dependencies in `package.json`)
  - [ ] Bug fixes (non-breaking change which fixes an issue)
  - [x] Improvements (misc. changes to existing features)
  - [x] Features (non-breaking change which adds functionality)

  ### My changes...
  - [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
  - [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
  - [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
  - [x] are not breaking changes.

  ### Documentation
  - [x] My changes do not require a change to the project documentation.
  - [ ] My changes require a change to the project documentation.
  - [ ] If yes to above: I have updated the documentation accordingly.

  **Note:** The changes are additive and automatically detect OAS 3.2 specs. Existing documentation covers the usage patterns, and OAS 3.2-specific documentation can be added in a follow-up PR once all features are complete.

  ### Automated tests
  - [ ] My changes can not or do not need to be tested.
  - [x] My changes can and should be tested by unit and/or integration tests.
  - [x] If yes to above: I have added tests to cover my changes.
  - [x] If yes to above: I have taken care to cover edge cases in my tests.
  - [x] All new and existing tests passed.

### CODE REVIEW

File: src/core/plugins/oas32/spec-extensions/wrap-selectors.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ * We wrap operationsWithRootInherited instead of operations because:
+ * - operations is a memoized createSelector that may be cached before our wrapper runs
+ * - operationsWithRootInherited consumes operations output and runs later in the chain
+ * - This ensures QUERY operations are added after the base operations are collected
+ *
+ * Reference: https://spec.openapis.org/oas/v3.2.0.html#path-item-object
+ */
+export const operationsWithRootInherited = (oriSelector, system) => {
+  return (state, ...args) => {
+    const isOAS32Selector = system.specSelectors.isOAS32
```
Comment (glowcloud) [inline src/core/plugins/oas32/spec-extensions/wrap-selectors.js]: Can we not wrap with `createOnlyOAS32SelectorWrapper` instead?

**Review by glowcloud — COMMENTED** [review]
Additional notes:
- We need to remove changes to `dist` folder before merging
- OAS32 plugin is not using JSON Schema 2020-12, so none of its keywords will be supported, this is how schemas should look like when its used:

Image

- Because we are not supporting 3.2 in Swagger Client yet, the dereferencing will be done as in 2.0/3.0. This means that dereferencing will not work correctly, e.g.:
```yaml
components:
   schemas:
      Test:
        type: object
        additionalProperties:
          type: string
      Test2:
        $ref: "#/components/schemas/Test"
        properties:
          testField2:
            type: string
```
after dereferencing will incorrectly be:
```yaml
components:
   schemas:
      Test:
        type: object
        additionalProperties:
          type: string
      Test2:
        additionalProperties:
          type: string
```

**Review by glowcloud — APPROVED** [review]

**Review by cka121 — APPROVED** [review]
